agent/tasks/executetestrun.go

```diff
@@ -104,27 +104,36 @@ func (ett ExecuteTestRunTask) Run() error {
 					log.Errorf("Failed to kill %s after timeout: %s", testletPath, err)
 				} else {
 					log.Debug("Successfully killed ", testletPath)
+
 				}
 			case err := <-done:
 				if err != nil {
 					log.Errorf("Testlet %s completed with error '%s'", testletPath, err)
-					gatheredData[thisTarget] = "error"
+
+					// TODO(mierdin): Handling testrun errors is on my plate as it is, and so
+					// this should get addressed properly in the future. The current approach
+					// of adding "error" to this map does nothing, as the status is tracked elsewhere
+					// gatheredData[thisTarget] = "error"
+
 				} else {
 					log.Debugf("Testlet %s completed without error", testletPath)
+
+					// Record test data
+					b := json.RawMessage(cmdOutput.Bytes())
+					gatheredData[thisTarget] = &b
```

Comment:
Style thing, I would `return` after logging the error and move the success case out of the else block. You could put a return in the timeout case too, then move the success down below the `select`. That way the success logic flows down the left side, making it easier to understand what the function does at a glance.